.github/reusable-audit.yml

```diff
@@ -0,0 +1,46 @@
+name: Reusable Audit Workflow
+
+on:
+  workflow_call:
+    inputs:
+      useForceResolutions:
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      WEBHOOK_E2E_FAILURE:
+        required: true
+
+jobs:
+  audit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.4.0
+      - uses: actions/setup-node@v2.5.1
+        with:
+          node-version: "16.x"
+          registry-url: "https://registry.npmjs.org"
+      - run: npm ci
+      # We check the dependency licenses as an audit task:
+      - run: npx license-checker --production --failOn "AGPL-1.0-only; AGPL-1.0-or-later; AGPL-3.0-only; AGPL-3.0-or-later; Beerware; CC-BY-NC-1.0; CC-BY-NC-2.0; CC-BY-NC-2.5; CC-BY-NC-3.0; CC-BY-NC-4.0; CC-BY-NC-ND-1.0; CC-BY-NC-ND-2.0; CC-BY-NC-ND-2.5; CC-BY-NC-ND-3.0; CC-BY-NC-ND-4.0; CC-BY-NC-SA-1.0; CC-BY-NC-SA-2.0; CC-BY-NC-SA-2.5; CC-BY-NC-SA-3.0; CC-BY-NC-SA-4.0; CPAL-1.0; EUPL-1.0; EUPL-1.1; EUPL-1.1;  GPL-1.0-only; GPL-1.0-or-later; GPL-2.0-only;  GPL-2.0-or-later; GPL-3.0; GPL-3.0-only; GPL-3.0-or-later; SISSL;  SISSL-1.2; WTFPL"
```

Comment:
License checking and auditing are two very different concerns that should not be run on the same schedule.

The success or failure of one should not affect the running of the other.

Reply:
Well, is there even a point checking dependencies that aren't allowed due to bad licenses for security issues? That is, this is just a quick gate; I think @NSeydoux suggested making this a audit check rather than a lint check.

Reply:
That's not the question.

The audit workflow should run on schedule and checks for dependencies vulnerabilities.

Gating on new dependencies that have problematic licenses should happen as a linting step on PRs.

That is, there is no point in checking licenses when running the scheduled audit check.

Reply:
A misconfigured or inappropriate dependency license is in the same category of issues as an NPM audit security failure. It _is_ a vulnerability.

Reply:
Correct me if I'm wrong, but in my understanding the semantics is the following: 
- linting verifies that our code adheres to good practices, is well-formatted, etc.
- auditing verifies that our dependencies are valid as per our policies, which includes both licensing and security.

It's not a formal nor strict definition, but that's the reasoning we had when moving the license checking as part of the audit workflow, since it doesn't apply to our code per se, but to our dependencies. I would say that whether the workflows run on schedule or on push is a slightly different question.

I 100% agree that we want the audit to run on a schedule as well as on push to detect security vulnerabilities as soon as possible when they are published, but I don't think it's too much of an issue that the license checking runs too in this context. If it's really problematic, we can always add a `if: github.event_name !== 'schedule'` to skip it.

Reply:
Any inappropriate dependency license would be introduced when a dependency is added or potentially updated (so a modified package-lock event) which means it will always happen via a PR/code change. I don't know what you mean by misconfigured.

Running an npm audit is addressing a very specific concern: "Make sure that no newly discovered vulnerability affects the specific versions of packages we depend on."; which happens overtime without code changes.

Reply:
I think it's more about communicating function, audit being "does our code meet our minimum security and legal requirements" is the way I'd think of it, not so much "does npm audit pass"